File: 160-intersection-of-two-linked-lists/160-intersection-of-two-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        if(headA==NULL || headB==NULL) return NULL;
        if(headA==headB) return headA;
        
        ListNode* ptr1=headA;
        ListNode* ptr2=headB;
        int countA=0,countB=0;
        
        while(ptr1!=NULL)
        {
            countA++;
            ptr1=ptr1->next;
        }
        while(ptr2!=NULL)
        {
            countB++;
            ptr2=ptr2->next;
        }
        int count=abs(countA-countB);
        cout<<countA<<' '<<countB;
        if(countA>countB)
        {
            while(count)
            {
                count--;
                headA=headA->next;
            }
        }
        else if(countB>countA)
        {
            while(count--)
            {
                headB=headB->next;
            }
        }
        while(headA!=NULL && headB!=NULL)
        {
            if(headA==headB) return headA;
            headA=headA->next;
            headB=headB->next;
        }
        return NULL;
    }
};

File: 160-intersection-of-two-linked-lists/160-intersection-of-two-linked-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */
var getIntersectionNode = function(headA, headB) {
    let countA=0
    let countB=0;
    let hA=headA;
    let hB=headB;
    while(hA!=null) {
        countA+=1;
        hA=hA.next;
    }
    while(hB!=null) {
        countB+=1;
        hB=hB.next;
    }
    
    uncommonNodes= Math.abs(countA-countB)
    
    if(countA>countB) {
        for(var i=0;i<uncommonNodes;i++) {
            headA=headA.next;
        }
        
    }
    else if(countB>countA) {
        for(var i=0;i<uncommonNodes;i++) {
            headB=headB.next;
        }
    }
    while(headA!=null && headB!=null) {
        if(headA==headB) {
            return headA;
        }
        else {
            headA=headA.next;
            headB=headB.next;
        }
    }
    return null;
};